### PR TITLE
Support universal links on iOS

### PR DESCRIPTION
**Pull request by Rexios80**
Supports universal links on iOS

The example app had a bunch of issues so I had to regenerate the iOS native code

### CODE REVIEW

File: flutter_web_auth_2/example/ios/Podfile.lock [hunk 1]
```diff
@@ -1,23 +1,28 @@
 PODS:
-  - Flutter (2.0.2)
-  - flutter_web_auth (0.4.1):
+  - Flutter (1.0.0)
+  - flutter_web_auth_2 (1.1.1):
```
Comment (ThexXTURBOXx) [inline flutter_web_auth_2/example/ios/Podfile.lock:3]: This seems rather odd to me, why does it not use the newest version here? Or is there something else wrongly set up or is this another Apple moment? :)

**Review by ThexXTURBOXx — COMMENTED** [review]
Thank you very much for your PR! There is one tiny thing I noticed. However that could be yet another Apple moment, maybe.

Also, one question about the Universal links: Could you maybe explain what they are? I am unsure since I have read about them online, but noone really explains the difference to normal deep links and custom registered schemes.

File: flutter_web_auth_2/example/ios/Podfile.lock (hunk 1 above)
Comment (Rexios80) [reply flutter_web_auth_2/example/ios/Podfile.lock:3]: These version numbers are based on the podspecs of the plugins which don't really mean anything. The podspec comes with the plugin and can be on whatever version and it doesn't matter.

Comment (Rexios80) [reply flutter_web_auth_2/example/ios/Podfile.lock:3]: Here's the documentation for universal links: https://developer.apple.com/library/archive/documentation/General/Conceptual/AppSearch/UniversalLinks.html

Major differences from custom url schemes:
- The links are just normal links
- You have to host a manifest on your domain to tell iOS what links it should open in your app
- It's "more secure" probably

Comment (ThexXTURBOXx) [reply flutter_web_auth_2/example/ios/Podfile.lock:3]: Okay, thank you very much for elaborating!
Have you tested your changes and if Universal links are now properly supported?
I could imagine that this check: https://github.com/ThexXTURBOXx/flutter_web_auth_2/blob/master/flutter_web_auth_2/lib/flutter_web_auth_2.dart#L35-L44
could forbid using them, but I am unsure.

Also, one last thing before merging: How does one use Universal links in `flutter_web_auth_2` now? Could you maybe add documentation for that in the README (e.g., add a new section [in Setup](https://github.com/ThexXTURBOXx/flutter_web_auth_2/blob/master/flutter_web_auth_2/README.md#setup))?

Comment (Rexios80) [reply flutter_web_auth_2/example/ios/Podfile.lock:3]: Universal links for OAuth redirects are working for me in my app. My concern is that I might have accidentally broken the existing method which I don't have a way to test easily.

You still need to pass a url scheme, but you just use `https`.

The setup for Universal Links might be too in depth to put in the readme. All that changes with how you use the plugin is passing `https` to the callbackUrlScheme. There is no special setup to use Universal Links with this plugin besides the normal universal link setup. I could just add that Universal Links are supported and link the Apple documentation if you want.

Comment (ThexXTURBOXx) [reply flutter_web_auth_2/example/ios/Podfile.lock:3]: Okay, passing `https` is plausible. Then, all the checks should work fine.
I hope that the old method still works. However, someone else might need to verify this.
Maybe @Knupper could verify this. But, if he does not have time, that's okay, too :)

Maybe there is a good link for the setup of Universal links? If passing `https` is the only thing that changes, I would probably only mention that and link to a setup guide for Universal links in general.

**Comment by ThexXTURBOXx** [comment]
Thank you very much! I will let this open until I (or rather someone else) can confirm that both the old method and Universal links work now :)

**Comment by rafalbednarczuk** [comment]
I tested the branch on IPhone 6s iOS 14.4 on:
--  `example` project from the repository with the default `foobar` schema
-- my own  project with `https` schema and universal links setup.

It works in both scenarios.

Adding universal links support to the iOS app is quite challenging, but here is the documentation:  
https://developer.apple.com/documentation/Xcode/supporting-associated-domains?language=objc

**Comment by ThexXTURBOXx** [comment]
@rafalbednarczuk Also here, thank you very much for testing and confirming!
I will merge and publish this PR as soon as I am home again.

**Comment by ThexXTURBOXx** [comment]
Again, thank you very much to the both of you! I have merged the PR and published version `2.0.3` with those changes included :)
Let me know if there is something that broke in the mean time! Otherwise, have fun with the new version :)

**Pull request merged**